### PR TITLE
feat(sdk): migrated cli func to apiV2

### PR DESCRIPTION
**Pull request by brohamgoham**

### CODE REVIEW

File: libs/goaccumulate/goaccumulate.go [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+	"github.com/AccumulateNetwork/accumulate/cmd/cli/db"
+	api2 "github.com/AccumulateNetwork/accumulate/internal/api/v2"
+	url2 "github.com/AccumulateNetwork/accumulate/internal/url"
+	"github.com/AccumulateNetwork/accumulate/protocol"
+	"github.com/mdp/qrterminal"
+)
+
+var (
+	Client         = client.NewAPIClient()
+	Db             db.DB
```
Comment (bunfield) [inline libs/goaccumulate/goaccumulate.go:22]: The database should not be in sdk. key management is handled outside the sdk.

File: libs/goaccumulate/goaccumulate.go [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+	TxPretend      = false
+)
+
+var (
+	BucketAnon     = []byte("anon")
+	BucketAdi      = []byte("adi")
+	BucketKeys     = []byte("keys")
+	BucketLabel    = []byte("label")
+	BucketMnemonic = []byte("mnemonic")
+)
```
Comment (bunfield) [inline libs/goaccumulate/goaccumulate.go:33]: no bucket definitions should be here

File: libs/goaccumulate/goaccumulate.go [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+		WhiteChar:      qrterminal.WHITE_WHITE,
+		WhiteBlackChar: qrterminal.WHITE_BLACK,
+		QuietZone:      2,
+	})
+
+	r, err := ioutil.ReadAll(b)
+	return string(r), err
+}
+
+func CreateAccount(url string, args []string) (string, error) {
```
Comment (bunfield) [inline libs/goaccumulate/goaccumulate.go:115]: the arguments passed to this function should not be an array of strings.  should be either the data structure of explicit parameters

File: libs/goaccumulate/goaccumulate.go [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+	}
+
+	binaryData, err := tac.MarshalBinary()
+	if err != nil {
+		return "", err
+	}
+
+	nonce := nonceFromTimeNow()
+	params, err := prepareGenTxV2(jsonData, binaryData, actor, si, privKey, nonce)
+	if err != nil {
```
Comment (bunfield) [inline libs/goaccumulate/goaccumulate.go:174]: we will not always have access to the private key.  For example, when passing on to a hardware wallet for signing.  need a signing callback or something similar

File: libs/goaccumulate/goaccumulate.go [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+		return "", err
+	}
+
+	var res api2.TxResponse
+
+	if err := Client.RequestV2(context.Background(), "create-token-account", params, &res); err != nil {
+		return PrintJsonRpcError(err)
+	}
+
+	return ActionResponseFrom(&res).Print()
```
Comment (bunfield) [inline libs/goaccumulate/goaccumulate.go:184]: the printing is customized for the cli and should not be part of the sdk.

File: libs/goaccumulate/goaccumulate.go [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+}
+
+func GenerateAccount() (string, error) {
+	return GenerateKey("")
+}
+
+func ListAccounts() (string, error) {
+
```
Comment (bunfield) [inline libs/goaccumulate/goaccumulate.go:192]: this function doesn't belong in the sdk.

File: libs/goaccumulate/goaccumulate.go (hunk 3 above)
Comment (bunfield) [reply libs/goaccumulate/goaccumulate.go:115]: perhaps explore the use of a builder pattern instead.

File: libs/goaccumulate/goaccumulate.go [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+		}
+		if lt.String() == string(v.Key) {
+			out += fmt.Sprintf("%s\n", v.Key)
+		}
+	}
+	return out, nil
+}
+
+func RestoreAccounts() (out string, err error) {
+	anon, err := Db.GetBucket(BucketAnon)
```
Comment (bunfield) [inline libs/goaccumulate/goaccumulate.go:212]: does not belong...

File: libs/goaccumulate/key.go [hunk 8]
```diff
@@ -0,0 +1,396 @@
+package goaccumulate
+
```
Comment (bunfield) [inline libs/goaccumulate/key.go:2]: key management should be handled by the wallet not the sdk.

File: libs/goaccumulate/page.go [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+		return "", fmt.Errorf("error retrieving key page for %s, %v", url, err)
+	}
+
+	res := api2.QueryResponse{}
+
+	err = json.Unmarshal(str, &res)
+	if err != nil {
+		return "", err
+	}
+	return PrintQueryResponseV2(&res)
```
Comment (bunfield) [inline libs/goaccumulate/page.go:25]: the return needs to be basic type or just json. the print is custom to the cli

File: libs/goaccumulate/page.go [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+	if err := Client.RequestV2(context.Background(), "create-key-page", params, &res); err != nil {
+		return PrintJsonRpcError(err)
+	}
+
+	ar := ActionResponse{}
+	//	err = json.Unmarshal(*res.Data, &ar)
+	//	if err != nil {
+	//		return "", fmt.Errorf("error unmarshalling create adi result, %v", err)
+	//	}
+	return ar.Print()
```
Comment (bunfield) [inline libs/goaccumulate/page.go:118]: see comments above about returning prints

**Review by bunfield — CHANGES_REQUESTED** [review]
instead of putting this in the libs directory, it should be under pkg (https://github.com/golang-standards/project-layout).  however, given this will require a lot more work, we should probably put it in internal/pkg until api is settled, at which time we can make it public.

I made comments to the first couple of files, but the comments carry throughout the code and apply to all files in this PR.

**Comment by firelizzard18** [comment]
Moving to GitLab

**Pull request closed without merging**